core/src/main/java/org/jivesoftware/spark/ui/login/MutualAuthenticationSettingsPanel.java

```diff
@@ -0,0 +1,265 @@
+                    PemHelper.saveToPemFile(request, IdentityController.CSR_FILE);
+                    JOptionPane.showMessageDialog(null, Res.getString("dialog.certificate.request.has.been.created") + IdentityController.SECURITY_DIRECTORY.toString());
+                } catch (OperatorCreationException | NoSuchAlgorithmException | IOException
+                        | NoSuchProviderException e1) {
+                    Log.error("Couldn't create Certificate Signing Request", e1);
+                }
+            } else if (selfSignedCertificate.isSelected()) {
+                try {
+                    KeyPair keyPair = idControll.createKeyPair();
+                    PemBuilder pemBuilder = new PemHelper().new PemBuilder();
+
+                    X509Certificate cert = idControll.createSelfSignedCertificate(idControll.createKeyPair());
+                    pemBuilder.add(keyPair);
+                    pemBuilder.add(cert);
+                    pemBuilder.saveToPemFile(IdentityController.CERT_FILE);
+                    JOptionPane.showMessageDialog(null, Res.getString("dialog.self.signed.certificate.has.been.created") + IdentityController.SECURITY_DIRECTORY.toString());
+                    
+
+                } catch (NoSuchAlgorithmException | NoSuchProviderException | IOException | OperatorCreationException
+                        | CertificateException e1) {
+                    Log.error("Couldn't create Self Signed Certificate", e1);
+                }
+            }
+        }        
+        
+    }
+
+    public static JTable getIdTable() {
+        return idTable;
+    }
+
+    @Override
+    public void mouseClicked(MouseEvent e) {
+        // TODO Auto-generated method stub
+        
+    }
+
+    @Override
+    public void mouseEntered(MouseEvent e) {
+        // TODO Auto-generated method stub
+        
+    }
+
+    @Override
+    public void mouseExited(MouseEvent e) {
+        // TODO Auto-generated method stub
+        
+    }
+
+    @Override
+    public void mousePressed(MouseEvent e) {
+        if (e.getClickCount() == 2) {
+            JTable source = (JTable) e.getSource();
+            if (e.getSource() == idTable) {
+                idControll.showCertificate();
+            }
+        }
+        if (e.getSource() == idTable) {
+            showCert.setEnabled(true);
+        }
+    }
+
+    @Override
+    public void mouseReleased(MouseEvent e) {
+
+    }
+
+    private void addCertificate() {
+
+        fileChooser.setAcceptAllFileFilterUsed(false);
+        fileChooser.addChoosableFileFilter(certFilter);
+        fileChooser.setFileFilter(certFilter);
+
+        int retVal = fileChooser.showOpenDialog(this);
+        if (retVal == JFileChooser.APPROVE_OPTION) {
+
+            File file = fileChooser.getSelectedFile();
+            try {
+                idControll.addEntryToKeyStore(file);
+                // idControll.addCertificateToKeystore(file);
+            } catch (IOException e) {
```

Comment:
Instead of writing this to a log file (or perhaps, in addition to writing it to a log file), it would be good to make an error message pop up. Most users don't look at the log file, and will not know that something went wrong.